### PR TITLE
MAINT: remove calc_lwork

### PR DESCRIPTION
**Pull request by argriffing**
closes https://github.com/statsmodels/statsmodels/issues/2159

### CODE REVIEW

File: statsmodels/tools/linalg.py [hunk 1]
```diff
@@ -75,11 +74,16 @@ def lstsq(a, b, cond=None, overwrite_a=0, overwrite_b=0):
...
-        v, x, s, rank, info = gelss(a1, b1, cond=cond, lwork=lwork,
-                                    overwrite_a=overwrite_a,
-                                    overwrite_b=overwrite_b)
+
+        # get optimal work array
+        work = gelss(a1, b1, lwork=-1)[4]
+        lwork = work[0].real.astype(np.int)
+        v, x, s, rank, work, info = gelss(
+            a1, b1, cond=cond, lwork=lwork, overwrite_a=overwrite_a,
+            overwrite_b=overwrite_b)
```
Comment (josef-pkt) [inline statsmodels/tools/linalg.py]: I like the old line breaks and formatting better

**Comment by josef-pkt** [comment]
Thank @argriffing  merged

**Comment by argriffing** [comment]
Sorry for changing the indentation style.

**Comment by josef-pkt** [comment]
> Sorry for changing the indentation style.

we will survive it :)

because it's a function that is essentially copied from scipy, I didn't mind if we stick with scipy formatting

**Pull request merged**